pkg/util/sets/string.go

```diff
@@ -150,6 +150,20 @@ func (s Set) SortedList() []string {
 	return res
 }
 
+// InsertContains inserts the item into the set and returns if it was already present.
+// Example:
+//
+//		if !set.InsertContains(item) {
+//			fmt.Println("Added item for the first time", item)
+//	  }
+func (s Set) InsertContains(item string) bool {
```

Comment:
any reason to not just have this be `Insert` with the same contract ? 

```
added := set.Insert(a)
if added {
...
}
```

Reply:
oh, i see. we used `Insert` already with a fluent-style return.

Reply:
yeah in theory we could switch it still and change current usages. However, I do worry slightly about if the contains() check is too slow for some use cases - some of the set usages are in hot paths

Reply:
i'd leave the existing `Insert` alone. No sense in confusing those that already know and use the package. I wish there was a pithier way of expressing the concept, but `CondInsert`, `TryInsert`, or even `MaybeInsert` don't seem any better to me. And I'm blanking on better ideas at the moment.

Reply:
nit: General name for this type of operation in other languages like Java is `AddIfAbsent` or `InsertIfAbsent`

Reply:
Its not quite the same IMO. Since it is a set there is no difference between just adding vs checking. The important part is the return.

Reply:
correct. InsertIfAbsent typically returns bool to say if it is added or not that indicates whether it is contained or not. https://www.geeksforgeeks.org/copyonwritearraylist-addifabsent-method-in-java/ but on list. 

But NBD, the current one is fine